infra/pr_helper.py

```diff
@@ -0,0 +1,224 @@
+
+
+def is_known_contributor(content, email):
+  """Checks if the author is in the contact list."""
+  return (email == content.get('primary_contact') or
+          email in content.get('vendor_ccs', []) or
+          email in content.get('auto_ccs', []))
+
+
+def has_author_modified_project(project_path, pr_author, headers):
+  """Checks if the author has modified this project before."""
+  commits_response = requests.get(
+      f'{BASE_URL}/commits?path={project_path}&author={pr_author}',
+      headers=headers)
+
+  if not commits_response.ok or not commits_response.json():
+    return False
+
+  commit = commits_response.json()[0]
+  return True, commit['sha']
+
+
+def get_pull_request_url(commit, headers):
+  """Gets the pull request url."""
+  pr_response = requests.get(f'{BASE_URL}/commits/{commit}/pulls',
+                             headers=headers)
+  if not pr_response.ok:
+    return None
+  return pr_response.json()[0]['html_url']
+
+
+def is_author_internal_member(pr_author, headers):
+  """Returns if the author is an internal member."""
+  response = requests.get(f'{BASE_URL}/contents/infra/MAINTAINERS.csv',
+                          headers=headers)
+  if not response.ok:
+    return False
+
+  maintainers = base64.b64decode(response.json()['content']).decode('UTF-8')
+  for line in maintainers.split('\n'):
+    print(f"username: {line.split(',')[2]}")
+    if pr_author == line.split(',')[2]:
+      save_env(None, None, True)
+      return True
+
+  return False
+
+
+def save_env(message, is_ready_for_merge, is_internal=False):
+  """Saves the outputs as environment variables."""
+  with open(os.environ['GITHUB_ENV'], 'a') as github_env:
+    github_env.write(f'MESSAGE={message}\n')
+    github_env.write(f'IS_READY_FOR_MERGE={is_ready_for_merge}\n')
+    github_env.write(f'IS_INTERNAL={is_internal}')
+
+
+def main():
+  """Verifies if a PR is ready for merge."""
+  token = os.environ['GITHUBTOKEN']
+  pr_author = os.environ['PRAUTHOR']
+  pr_number = os.environ['PRNUMBER']
+  os.environ['GITHUB_AUTH_TOKEN'] = token
+  headers = {
+      'Authorization': f'Bearer {token}',
+      'X-GitHub-Api-Version': '2022-11-28'
+  }
+  message = ''
+  is_ready_for_merge = True
+
+  # Bypasses PRs of the internal members.
+  if is_author_internal_member(pr_author, headers):
+    return
+
+  # Gets all modified projects path.
+  projects_path = get_projects_path(pr_number, headers)
+  email = get_author_email(pr_number, headers)
+
+  for project_path in projects_path:
+    project_url = f'{GITHUB_URL}/{OWNER}/{REPO}/tree/{BRANCH}/{project_path}'
+    content_dict = get_project_yaml(project_path, headers)
+
+    # Gets information for the new integrating project
+    if not content_dict:
+      is_ready_for_merge = False
+      new_project = get_integrated_project_info(pr_number, headers)
+      repo_url = new_project.get('main_repo')
+      if repo_url is not None:
+        message += (f'@{pr_author} is integrating a new project:<br/>'
```

Comment:
I think we should still add a message even if the repo_url is not set. Just say it's missing and can't compute criticality score.